facade/service.go

```diff
@@ -176,32 +176,90 @@ func (f *Facade) MigrateService(ctx datastore.Context, request dao.ServiceMigrat
 		}
 	}
 
-	svcs, err = readNewServiceDefinitions(outputFileName)
+	svcsMap, err := readNewServiceDefinitions(outputFileName)
 	if err != nil {
 		return err
 	}
 
-	for _, svc := range svcs {
-		if request.DryRun {
-			err = f.verifyServiceForUpdate(ctx, svc, nil)
-			if err == nil {
-				glog.V(2).Infof("Facade:MigrateService: dry-run of migration script complete for serviceID %+v", svc.ID)
+	// Validate the modified services.
+	for _, svc := range svcsMap["modified"] {
+		if err = f.verifyServiceForUpdate(ctx, svc, nil); err != nil {
+			return err
+		}
+	}
+
+	// Validate the cloned services.
+	existing, err := f.serviceStore.GetServices(ctx)
```

Comment:
This function is getting really long. Can you refactor lines 192-213 into a private method, something like validateClonedServices?